include/sound/soc-dpcm.h

```diff
@@ -61,6 +61,22 @@ enum snd_soc_dpcm_trigger {
 	SND_SOC_DPCM_TRIGGER_BESPOKE,
 };
 
+/**
+ * enum snd_soc_update_mode - mode for calling soc_dpcm_runtime_update()
+ *
+ * @SND_SOC_UPDATE_RUNTIME:	default mode, used for mux, mixer, and volume
+ *				widgets
+ * @SND_SOC_UPDATE_STARTUP:	a pipeline, partially owned by a virtual
+ *				machine, is starting
+ * @SND_SOC_UPDATE_SHUTDOWN:	a pipeline, partially owned by a virtual
+ *				machine, is shutting down
```

Comment:
@plbossart volume on the host is never changed by the guest. The only times when these are used on behalf of a guest are when starting or stopping a pipeline. I'll work on addressing the rest in the design document and in this commit.

Reply:
I tried to get to the bottom of this. The actual goal of this was to enforce suitable for VirtIO life cycles for stream / stop, modprobe / rmmod, reboot, boot / halt. With the current configuration all those cases were tested to work. But some of the changes, that I had to perform were derived in a rather empiric way. This is one of them - I found a way to enforce a functioning life cycle for the guest, although it might not be the right way to do that, but I haven't so far been able to understand how it's supposed to work, and I don't have any working test case to exercise any `soc_dpcm_runtime_update()` paths to see how they are supposed to work. So, I guess, we'll have to send what I have now and see if we get any good ideas for how to implement this correctly?

Reply:
@lyakh is the goal here to run ```soc_dpcm_runtime_update()``` outwith the existing DPCM flow ?

Reply:
@lgirdwood that's how I understood your suggestion - to use `snd_soc_runtime_(de)activate()` and `soc_dpcm_runtime_update()` when starting and stopping guest streaming, yes.

Reply:
Ok, so making soc_dpcm_runtime_* APIs callable outside of the DPCM flow is the aim here and making it generic (not tied to virtio use case with Virtio strings) is the way to go. Please send this as a new PR, we can then review before mailing to the list.